docs/pages/lighting-setups.rst

```diff
@@ -0,0 +1,197 @@
+
+:summary: This tutorial demonstrates the creation and manipulation of LightSetups in Habitat-sim.
+
+.. contents::
+    :class: m-block m-default
+
+A **LightSetup** consists of a set of LightInfo structures defining the common configuration
+of a set of point lights used to render objects in a scene. Once defined and registered,
+a LightSetup can be assigned to any subset of objects in the scene, including the scene asset itself.
+
+Each **LightInfo** structure in a LightSetup defines the `color`, `position`, and `LightPositionModel` of a single point light source.
+The LightPositionModel defines the coordinate frame of the light and can be set to any of:
+
+.. code:: python
+
+    habitat_sim.gfx.LightPositionModel.CAMERA
+    habitat_sim.gfx.LightPositionModel.GLOBAL
+    habitat_sim.gfx.LightPositionModel.OBJECT
+
+Each LightSetup is registered in the simulator via a unique key. Two default lighting setups are pre-defined:
+
+
+- **habitat_sim.gfx.DEFAULT_LIGHTING_KEY**
+
+  8 white, GLOBAL lights at scene bounding box corners.
+
+- **habitat_sim.gfx.NO_LIGHT_KEY**
+
+  0 lights (for Flat shaded assets).
+
+
+Additional custom setups can be created and registered via the **simulator.set_light_setup** function:
+
+.. code:: python
+
+    sim.set_light_setup(new_light_setup, "my_custom_lighting_key")
+
+Any existing LightSetup can be queried with **simulator.get_light_setup**:
+
+.. code:: python
+
+    custom_light_setup = sim.set_light_setup(n"my_custom_lighting_key")
+
+
+`Working with Lights Example:`_
+===============================
+
+The example code below demonstrates the usage of default and custom LightSetups and is runnable via:
+
+.. code:: shell-session
+
+    $ python examples/tutorials/lighting_tutorial.py
+
+First, we import modules we will need, define some convenience functions, and initialize the Simulator, Agent, and objects.
+
+.. include:: ../../examples/tutorials/lighting_tutorial.py
+    :code: py
+    :start-after: # [setup]
+    :end-before: # [/setup]
+
+`Scene Lighting`_
+=================
+
+By default, the scene will be shaded with no lights using the **habitat_sim.gfx.NO_LIGHT_KEY** LightSetup.
+This configuration is ideal for scene assets with illumination baked into textures (e.g. building scans such as MP3D).
+
+.. include:: ../../examples/tutorials/lighting_tutorial.py
+    :code: py
+    :start-after: # [default scene lighting]
+    :end-before: # [scene swap shader]
+
+.. image:: images/lighting-setups-images/0.jpg
+    :width: 45%
+
+To use a custom light setup for the scene, edit the **habitat_sim.SimulatorConfiguration.scene_light_setup**
+option when creating/reconfiguring your Simulator. Note that
+while you can modify the scene's LightSetup dynamically during runtime, you will need to
+reconfigure the simulator to switch the scene's LightSetup key. Also, due to asset loading specifics
+**you must close and re-initialize the Simulator to swap between Flat and Phong shading setups**.
```

Comment:
Ouch :-(, but beyond the scope of tutorial.

Reply:
Yes, but we are choosing whether to compile normals and how to load the material data at asset load time. I agree this needs to be addressed. I wrote this section to avoid future issues opening without a place to point to. 😅

Reply:
Right, normals, forgot about those.... tho for models that are designed to be shadeless/unlit this could be forced to be always Flat, no? So then you could easily flip between Phong and Flat for models that *can* be shaded, while having Flat always for those which can't.

But yes, this is out of scope of this tutorial. Let's revisit that later.

Reply:
Hi everyone :) thought I'd check in on my pending PRs to make sure things were going smoothly. Just wanted to clarify here that you **can** switch between flat/phong by requiring normals at load time and **correctly** specifying ambient color/texture. The issue is many of our models use a material model which isn't supported (PBR), so ambient color/texture isn't specified. Therefore, this issue is fixed without any changes to the lighting code, but instead

- supporting more material models (PBR should cover most cases)
- material manipulation, so users can swap diffuse <-> ambient textures if they really don't want ambient present during phong shading

Reply:
Thanks for your input here, @matthewjmay. 

I agree with your proposed solutions. Material editing would give us some options for unblocking this use-case, and we are actively considering how to move forward on this. PBR support is also on our radar. 

For now, this documented work-around is functional, but not ideal.